### DISCUSSION TITLE
maxAttempts in Jedis Cluster constructor

### DISCUSSION THREAD

**Discussion by ashishkeshu** [xetorthio/jedis#2624] [Q&A]
Hi Team,

I am using below constructor to create jedis cluster.
new JedisCluster(jedisClusterNode, connTimeout, soTimeout, **maxAttempts**, username, password, "", jedisPoolConfig, false);

what is the purpose of maxAttempts here in this constructor, in which scenario it is useful to set and what is optimized value for this.

Thanks,
Ashish

**Comment by sazzad16** [comment]
Due to slot migration and many other reasons, (master) node for specific slot may be unavailable at a certain moment. JediCluster is configured to retry a command in those moments may have a better chance to be executed. This configuration is controlled by maxAttempts parameter. If you don't want to retry at all, just set `maxAttempts=1`.

**Reply by ashishkeshu** [reply]
Hi Sazzad,

Thanks for your response.
can you please tell the role of maxAttempts while making connection with redis server.
in case all the redis server nodes are unreachable for a that moment then will it reattempt to establish the connection with redis server based on the maxAttempts values?

Thanks in advance.

**Reply by gyula-lakatos** [reply]
@sazzad16 I'm fighting with the same thing. Can I configure something like retry with exponential backoff as long as the app is unable to connect to Redis? What happens if I write a negative value to maxAttempts? It will retry indefinitely? Or can I write Long.MAX_VALUE to achieve the same functionality? It might be only me, but I see no documentation for maxAttempts anywhere. :/

**Reply by sazzad16** [reply]
@gyula-lakatos Due to the presence of `maxTotalRetriesDuration` parameter, it will not try indefinitely. You may try combination of different values for maxAttempts and maxTotalRetriesDuration.
